notify/notify_test.go

```diff
@@ -981,35 +1006,59 @@ func TestTimeActiveStage(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			r := prometheus.NewRegistry()
+			marker := types.NewMarker(r)
 			metrics := NewMetrics(r, featurecontrol.NoopFlags{})
 			intervener := timeinterval.NewIntervener(test.intervals)
-			st := NewTimeActiveStage(intervener, metrics)
+			st := NewTimeActiveStage(intervener, marker, metrics)
 
 			// Get the names of all time intervals for the context.
 			activeTimeIntervalNames := make([]string, 0, len(test.intervals))
 			for name := range test.intervals {
 				activeTimeIntervalNames = append(activeTimeIntervalNames, name)
 			}
+			// Sort the names so we can compare mutedBy with test.mutedBy.
+			sort.Strings(activeTimeIntervalNames)
 
 			ctx := context.Background()
 			ctx = WithNow(ctx, test.now)
+			ctx = WithGroupKey(ctx, "group1")
 			ctx = WithActiveTimeIntervals(ctx, activeTimeIntervalNames)
 			ctx = WithMuteTimeIntervals(ctx, nil)
+			ctx = WithRouteID(ctx, "route1")
 
 			_, active, err := st.Exec(ctx, log.NewNopLogger(), test.alerts...)
 			require.NoError(t, err)
 
 			if len(test.mutedBy) == 0 {
 				// All alerts should be active.
 				require.Equal(t, len(test.alerts), len(active))
+				// The group should not be marked.
+				mutedBy, isMuted := marker.Muted("route1", "group1")
+				require.False(t, isMuted)
+				require.Empty(t, mutedBy)
 				// The metric for total suppressed notifications should not
 				// have been incremented, which means it will not be collected.
-				require.NoError(t, prom_testutil.GatherAndCompare(r, strings.NewReader("")))
+				require.NoError(t, prom_testutil.GatherAndCompare(r, strings.NewReader(`
+# HELP alertmanager_marked_alerts How many alerts by state are currently marked in the Alertmanager regardless of their expiry.
+# TYPE alertmanager_marked_alerts gauge
+alertmanager_marked_alerts{state="active"} 0
```

Comment:
Do I need to add metrics for marked groups? @gotjosh

Reply:
I don't think so - I wouldn't really understand their historically use-case.